azurecontainerapps/azurecontainerapps.ts

```diff
@@ -30,200 +23,507 @@ export class azurecontainerapps {
+        }
+
+        // Ensure that one of appSourcePath, imageToDeploy, or yamlConfigPath is provided
+        if (util.isNullOrEmpty(this.appSourcePath) && util.isNullOrEmpty(this.imageToDeploy) && util.isNullOrEmpty(this.yamlConfigPath)) {
+            tl.error(tl.loc('MissingRequiredArgumentMessage'));
+            throw Error(tl.loc('MissingRequiredArgumentMessage'));
+        }
+    }
+
+    /**
+     * Sets up the Azure CLI to be used for this task by logging in to Azure with the provided service connection and
+     * setting the Azure CLI to dynamically install missing extensions.
+     */
+    private static setupAzureCli() {
+        // Log in to Azure with the service connection provided
+        const connectedService: string = tl.getInput('connectedServiceNameARM', true);
+        this.authHelper.loginAzureRM(connectedService);
+
+        // Set the Azure CLI to dynamically install missing extensions
+        util.setAzureCliDynamicInstall();
+    }
+
+    /**
+     * Sets up the resources required to deploy a Container App. This includes the following:
+     * - Getting or generating the Container App name
+     * - Getting or discovering the location to deploy resources to
+     * - Getting or creating the resource group
+     * - Getting or creating the Container App Environment
+     */
+    private static setupResources() {
+        // Get the Container App name if it was provided, or generate it from build variables
+        this.containerAppName = this.getContainerAppName();
+
+        // Get the location to deploy resources to, if provided, or use the default location
+        this.location = this.getLocation();
+
+        // Get the resource group to deploy to if it was provided, or generate it from the Container App name
+        this.resourceGroup = this.getOrCreateResourceGroup(this.containerAppName, this.location);
+
+        // Determine if the Container App currently exists
+        this.containerAppExists = this.appHelper.doesContainerAppExist(this.containerAppName, this.resourceGroup);
+
+        // If the Container App doesn't exist, get/create the Container App Environment to use for the Container App
+        if (!this.containerAppExists) {
+            this.containerAppEnvironment = this.getOrCreateContainerAppEnvironment(this.containerAppName, this.resourceGroup, this.location);
+        }
+    }
+
+    /**
+     * Gets the name of the Container App to use for the task. If the 'containerAppName' argument is not provided,
+     * then a default name will be generated in the form 'ado-task-app-<buildId>-<buildNumber>'.
+     * @returns The name of the Container App to use for the task.
+     */
+    private static getContainerAppName(): string {
+        let containerAppName: string = tl.getInput('containerAppName', false);
+        if (util.isNullOrEmpty(containerAppName)) {
+            containerAppName = `ado-task-app-${this.buildId}-${this.buildNumber}`;
```

Comment:
The default build number of `$(Date:yyyyMMdd).$(Rev:r)` ([docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/run-number?view=azure-devops&tabs=yaml)) contains a period, which I don't think is allowed to be in a container app name, according to [its naming conventions](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftapp). 
```
Max length: 32
Lowercase letters, numbers, and hyphens.. Start with letter and end with alphanumeric.
```